include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -207,6 +207,14 @@ constexpr __target __target_device =
     __target::global_buffer;
 #endif
 
+template <typename DataT>
+using __buffer_allocator =
+#if __LIBSYCL_VERSION >= 50707 || defined(SYCL2020_CONFORMANT_APIS)
```

Comment:
When is the macro `SYCL2020_CONFORMANT_APIS` defined? Should we possibly use it for other definitions in this file/namespace as well?

Reply:
Previously this define was used in SYCL headers which was :

// Guard SYCL 2020 buffer_allocator with template arguments behind the
// SYCL2020_CONFORMANT_APIS macro.
#ifdef SYCL2020_CONFORMANT_APIS
template <typename DataT>
using buffer_allocator = detail::sycl_memory_object_allocator<DataT>;
#else
using buffer_allocator = detail::sycl_memory_object_allocator<char>;
#endif

In new compiler versions and SYCL library this define already removed from code.

I think that we may avoid to use them.
Fixed.